src/checkDifferences.ts

```diff
@@ -18,104 +104,49 @@ export const checkDifferences = async (shotItems: ShotItem[]) => {
-
-  const comparisonResults: Record<
-    string,
-    {
-      pixelDifference: number;
-      pixelDifferencePercentage: number;
-      isWithinThreshold: boolean;
-    }
-  > = {};
+  const differences: Differences = {
+    aboveThresholdDifferenceItems: [],
+    comparisonResults: {},
+    noBaselinesItems: [],
+  };
 
   await mapLimit<[number, ShotItem], void>(
     shotItems.entries(),
     config.compareConcurrency,
     async (item: [number, ShotItem]) => {
-      const [index, shotItem] = item;
-      const logger = (message: string) => {
-        log
-          .item({
-            shotMode: shotItem.shotMode,
-            uniqueItemId: shotItem.shotName,
-            itemIndex: index,
-            totalItems: total,
-          })
-          .process('info', 'general', message);
-      };
-
-      logger(`Comparing '${shotItem.id}'`);
-
-      const baselineImageExists = existsSync(shotItem.filePathBaseline);
-
-      if (!baselineImageExists) {
-        logger('Baseline image missing. Will be treated as addition.');
-        noBaselinesItems.push(shotItem);
-
-        return;
-      }
-
-      const currentImageExists = existsSync(shotItem.filePathCurrent);
-
-      if (!currentImageExists) {
-        throw new Error(
-          `Error: Missing current image: ${shotItem.filePathCurrent}`,
-        );
-      }
-
-      const { pixelDifference, pixelDifferencePercentage, isWithinThreshold } =
-        await compareImages(
-          shotItem.threshold,
-          shotItem.filePathBaseline,
-          shotItem.filePathCurrent,
-          shotItem.filePathDifference,
-        );
-
-      if (shallGenerateMeta()) {
-        comparisonResults[shotItem.id] = {
-          pixelDifference,
-          pixelDifferencePercentage,
-          isWithinThreshold,
-        };
-      }
-
-      if (pixelDifference > 0) {
-        const percentage = (pixelDifferencePercentage * 100).toFixed(2);
-
-        if (isWithinThreshold) {
-          logger(
-            `Difference of ${pixelDifference} pixels (${percentage}%) found but within threshold.`,
-          );
-        } else {
-          aboveThresholdDifferenceItems.push(shotItem);
-          logger(
-            `Difference of ${pixelDifference} pixels (${percentage}%) found. Difference image saved to: ${shotItem.filePathDifference}`,
-          );
-        }
-      } else {
-        logger('No difference found.');
-      }
+      const difference = await checkDifference({
+        item,
+        total: shotItems.length,
+      });
+
+      addDifferenceToDifferences({
+        difference,
+        differences,
+        shotItem: item[1]
+      })
     },
   );
 
-  if (shallGenerateMeta()) {
-    log.process(
-      'info',
-      'general',
-      `Writing meta file with ${
-        Object.entries(comparisonResults).length
-      } items.`,
-    );
-    writeFileSync(
-      `${path.join(config.imagePathCurrent, 'meta')}.json`,
-      JSON.stringify(comparisonResults, null, 2),
-    );
-  }
-
```

Comment:
Moved to here: https://github.com/lost-pixel/lost-pixel/pull/385/files#diff-d64beacb5711e9646c8e8e25cf23f283358d86c7f5f50b6f3aaf18044e356942R90

so it also applies to differences from `createShots`